### PR TITLE
NAS-128631 / 24.10 / Make sure we redirect logs to relevant log files instead of copying them

### PR DESCRIPTION
**Pull request by Qubad786**
## Problem

We have different log filters specified where we want to make sure that different sources i.e k3s/scst etc do not pollute other log files with their messages but unfortunately what in reality is happening is that we have their logs being logged to separate files but they are still getting logged in other log files i.e daemon.log etc which indicates that logs are being copied rather then redirected completely.

## Solution

Add changes to make sure we don't process other log filters if we have a match for a filter earlier and once we have gone through our custom log file solutions we can finally have rest of the filters for other log files being processed.

### CODE REVIEW

**Comment by anodos325** [comment]
Have you verified that the logs still get forwarded correctly to remote syslog server if you configure one?

**Comment by Qubad786** [comment]
> Have you verified that the logs still get forwarded correctly to remote syslog server if you configure one?

@anodos325 the logs are still being forwarded as they were before so we are good on that end

File: src/freenas/etc/logrotate.d/netdata_override [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/var/log/netdata/*.log {
```
Comment (sonicaj) [inline src/freenas/etc/logrotate.d/netdata_override:1]: Would this override only specific logging configuration netdata package itself provides or is like a complete override of what's specified here and original provided config is retained if that config options are not specified here ?

Comment (Qubad786) [reply src/freenas/etc/logrotate.d/netdata_override:1]: Yes so for example if `abcd` is specified in original config and is not overridden here, that will be retained when the configuration is applied as a whole ✅

**Review by sonicaj — APPROVED** [review]

**Review by anodos325 — APPROVED** [review]

**Pull request merged**